jnim/private/jni_export.nim

```diff
@@ -399,11 +401,17 @@ macro jexport*(a: varargs[untyped]): untyped =
 
         thunkCall = newCall(bindSym"nimValueToJni", envName, thunkCall, copyNimTree(thunkParams[0]))
 
-        let thunk = newProc(thunkName, thunkParams)
+        let thunk = newProc(postfix(thunkName, "*"), thunkParams)
         thunk.addPragma(newIdentNode("cdecl"))
+        thunk.addPragma(newIdentNode("dynlib"))
         thunk.addPragma(newIdentNode("exportc")) # Allow jni runtime to discover the functions
         thunk.body = quote do:
+          # TODO: somehow pair this with a matching tearDownForeignThreadGC when necessary
+          setupForeignThreadGC()
           if theEnv.isNil: theEnv = `envName`
+          let `thisRef` = theEnv.jniObjectToNimObj(this, type(`classNameIdent`))
+          GC_ref(`thisRef`)
```

Comment:
This should not be necessary. `thisRef` always outlives `thinkCall` as it is passed into it.

Reply:
I don't understand why, but I got crashes that looked like results of Nim GC disposing of "this" in the middle of a long-running `thunkCall` function (in a Java-originating thread). After I added GC_disable as a quick test, they disappeared. Adding this GC_ref also resolved them. Unfortunately I have no idea how to find out if a variable is considered referenced by Nim GC or not.

Reply:
Hm... Sorry, that doesn't cut it :). Since we both know GC_ref is not needed here, I assume there might be a bug elsewhere which happens to be hidden by GC_ref.

Reply:
That's a tough one :/ I will remove this commit for now from this PR then, and maybe try to build an issue report with some more details so that we can hopefully discuss and think of ideas, what do you think?

Reply:
Give me some time to play around with this pr as is